tests/conditional_test.py

```diff
@@ -119,20 +133,28 @@ def test_pytket_qir_conditional_v() -> None:
     circ.add_gate(OpType.H, [0], condition_bits=[0, 1, 2], condition_value=3)
 
     result = pytket_to_qir(
-        circ, name="test_pytket_qir_conditional_v", qir_format=QIRFormat.STRING
+        circ,
+        name="test_pytket_qir_conditional_v",
+        qir_format=QIRFormat.STRING,
     )
 
+    check_qir_result(result, "test_pytket_qir_conditional_v")
+
+
+def test_pytket_qir_conditional_5() -> None:
+    # test complicated conditions and recursive classical op
+
     circ = Circuit(2, 2).H(0).H(1).measure_all()
 
     circ.add_gate(OpType.H, [0], condition_bits=[0, 1], condition_value=3)
 
-    result_2 = pytket_to_qir(
-        circ, name="test_pytket_qir_conditional_v", qir_format=QIRFormat.STRING
+    result = pytket_to_qir(
+        circ,
+        name="test_pytket_qir_conditional_5",
+        qir_format=QIRFormat.STRING,
     )
 
-    check_qir_result(result, "test_pytket_qir_conditional_v")
-
-    check_qir_result(result_2, "test_pytket_qir_conditional_v")
```

Comment:
Also refactored this test into its own test: `test_pytket_qir_conditional_5`

Reply:
So they are not checked to be identical anymore? What it the reason for that?

Reply:
Because the first circuit is `Circuit(2, 3)` and the second `Circuit(2, 2)`, ie, they have different sizes for the classical register.